cookbooks/ros2_windows/recipes/visual_studio.rb

```diff
@@ -30,7 +30,16 @@
 
 installer_options = base_options + package_arguments
 
-visual_studio_source = 'https://aka.ms/vs/16/release/vs_%s.exe' % node['ros2_windows']['vs_version']
+case node['ros2_windows']['vs_release']
+when '2022'
+  visual_studio_source = 'https://aka.ms/vs/17/release/vs_%s.exe' % node['ros2_windows']['vs_version']
+  visual_studio_path = 'c:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\%s' % vs_version_camel_case
+when '2019' 
+  visual_studio_source = 'https://aka.ms/vs/16/release/vs_%s.exe' % node['ros2_windows']['vs_version']
+  visual_studio_path = 'c:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\%s' % vs_version_camel_case
+else 
+  raise 'Unsupported Visual Studio version' % node['ros2_windows']['vs_release']
```

Comment:
There was no string expansion in your format string so this would have been just the string literal (or an error, I forget how Ruby handles format arity)

```suggestion
  raise "Unsupported Visual Studio version: #{node['ros2_windows']['vs_release']}"
```